src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -172,6 +161,47 @@ void UprobesUnwindingVisitor::visit(CallchainSamplePerfEvent* event) {
     return;
   }
 
+  // The leaf function is not guaranteed to have the frame pointer for all our targets. Though, we
+  // assume that $rbp remains untouched by the leaf functions, such that we can rely on
+  // perf_event_open to give us "almost" correct callstacks (the caller of the leaf function will be
+  // missing). We do a plausibility check for this assumption by checking if the callstack only
+  // contains executable code.
+  // TODO(b/187690455): As soon as we actually have frame pointers in all non-leaf functions, we can
+  //  "discard" the sample. Till that point this check will always fail, because the caller of
+  //  __libc_start_main will be invalid since libc doesn't have frame-pointers.
+  //  Note that, that at this point in time, we don't want to actually throw the sample away, but
+  //  rather report it as "broken".
+  for (uint64_t frame_index = 1; frame_index < event->GetCallchainSize(); ++frame_index) {
+    unwindstack::MapInfo* map_info = current_maps_->Find(event->GetCallchain()[frame_index]);
+    if (map_info == nullptr || (map_info->flags & PROT_EXEC) == 0) {
+      ERROR(
```

Comment:
This could spam the logs I believe.

Reply:
Yes, at least until we have frame pointers in glibc. Otherwise, we will end for all frame-pointer based samples in being unwinding errors. That would be a large burden for us, as we could not test the frame-pointer code anymore. Same will be probably true for our integration tests, though.

Reply:
Ah, so basically this is the main place where we check if frame pointer unwinding was successful, and we want to get wrong but complete callstacks (instead of the unwind error) for now for experimental purposes. Is that correct? In that case that's fine but maybe that should be clarified in the comment.
I'm still wondering whether the `ERROR` is spammy when experimenting but then that's for you to say.

Reply:
I removed the error and changed the comment. Left the rest of the code as is for now.